apps/ui/src/components/Modal/Vote.vue

```diff
@@ -88,6 +88,16 @@ watch(
   { immediate: true }
 );
 
+watch(
+  () => props.proposal,
+  (toProposal, fromProposal) => {
+    if (props.open && fromProposal?.id !== toProposal.id) {
+      emit('close');
+    }
+  },
+  { immediate: true }
```

Comment:
This is a bit weird with `immediate` flag. Right now if we just mount it with open flag directly it will close itself (because `fromProposal` will be undefined, because it's its first mount).

I think it should be not immediate.

Reply:
This was to fix a bug, where the modal will stay open when changing proposal by navigating back with the browser back button.

But not needed anymore with the fix above, closing the modal from the calling component, instead of watching inside the modal itself